IN-CORE/pyincore
# 71 r3 refactoring to expressions

Changes made:
- Added method to calculate the inverse of cdf which can be used in repair curve calculation to calculate repair time when percent of functionality is provided as an input. Used this in residentialbuildingrecovery analysis
- Changed the DFR3 expression evaluation method to also accept an array of numeric values, instead of a single value. This improves analyses performance where an array of x values can be evaluated together. 
- Added support for Restoration curves in pyincore. Included pytests.
- Updated the format of repair curves needed for residentialbuildingrecovery in DEV DB. Converted them to be stored as CDF instead of PPF

Things to test:
- See the new pytests included in this PR - repairs, restorations, evaluating curve expressions and its inverse
- Make sure residentialbuildingrecovery is not impacted due to repair curve change
- Make sure none of the fragility based analyses were impacted due to the changes made to `solve_curve_expression()`

Closes #71
